sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java

```diff
@@ -30,23 +32,29 @@
  * The result of a {@link BigQueryIO.Write} transform.
  */
 public final class WriteResult implements POutput {
-
   private final Pipeline pipeline;
+  private final TupleTag<TableRow> failedInsertsTag = new TupleTag<>();
+  private final PCollection<TableRow> failedInserts;
```

Comment:
I thought about it, but I had a trouble imagining a good use case. If we did, I'd prefer FailedInsertRow over a KV.